src/pharmpy/model/external/nonmem/advan.py

```diff
@@ -590,36 +593,102 @@ def _advan12_trans(trans: str):
 
 
 def dosing(di: DataInfo, dataset, dose_comp: int):
+    # Only check doses
+    if dataset is not None:
+        dataset = dataset[dataset['AMT'] != 0]
+
     if 'CMT' not in di.names or di['CMT'].drop or dataset is None:
-        return ({'comp_number': dose_comp, 'dose': _dosing(di, dataset, dose_comp)},)
+        if 'ADMID' in di.names:
```

Comment:
Yes, that is about the functionality of the function. But instead of only having different subsets, the function will loop through different parts of the dataset (such as the CMT column or the admid column). I have reformatted the structure for readability but please let me know if there are any issues :)